logstash-core/lib/logstash/settings.rb

```diff
@@ -0,0 +1,194 @@
+# encoding: utf-8
+
+module LogStash
+  class Settings
+
+    def initialize
+      @settings = {}
+    end
+
+    def register(setting)
+      if @settings.key?(setting.name)
+        raise ArgumentError.new("Setting \"#{setting.name}\" has already been registered as #{setting.inspect}")
```

Comment:
but also this might work, as the context is already set to settings, it all depends who might be the consumer of this exception.